Assets/Scripts/SS3D/Systems/Tile/Connections/AdjacencyMap.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using FishNet.Object.Synchronizing;
+using FishNet.Object.Synchronizing.Internal;
+using SS3D.Systems.Tile.Connections.AdjacencyTypes;
+
+namespace SS3D.Systems.Tile.Connections
+{
+    /// <summary>
+    /// Used for storing adjacencies.
+    /// </summary>
+    public class AdjacencyMap
+    {
+        //Stores an array of which of the 8 surrounding tiles have a connection. Order assumed to match Direction enum values.
+        private AdjacencyData[] _connections;
+
+        public AdjacencyMap()
+        {
+            _connections = new [] {
+                new AdjacencyData (TileObjectGenericType.None, TileObjectSpecificType.None, false),
+                new AdjacencyData (TileObjectGenericType.None, TileObjectSpecificType.None, false),
+                new AdjacencyData (TileObjectGenericType.None, TileObjectSpecificType.None, false),
+                new AdjacencyData (TileObjectGenericType.None, TileObjectSpecificType.None, false),
+                new AdjacencyData (TileObjectGenericType.None, TileObjectSpecificType.None, false),
+                new AdjacencyData (TileObjectGenericType.None, TileObjectSpecificType.None, false),
+                new AdjacencyData (TileObjectGenericType.None, TileObjectSpecificType.None, false),
+                new AdjacencyData (TileObjectGenericType.None, TileObjectSpecificType.None, false),
+            };
+        }
+
+        public AdjacencyData[] Connections
+        {
+            get => _connections;
+            set => _connections = value;
```

Comment:
that set should not accept just anything, otherwise there's no use for having a private array in the first place.